src/components/ReportActionItem/ReportPreview.tsx

```diff
@@ -511,6 +507,95 @@ function ReportPreview({
         Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(iouReportID));
     }, [iouReportID]);
 
+    const reportPreviewAction = useMemo(() => {
+        if (!iouReport) {
+            return CONST.REPORT.REPORT_PREVIEW_ACTIONS.VIEW;
+        }
+        return getReportPreviewAction(iouReport, transactions, violations, policy);
+    }, [iouReport, policy, transactions, violations]);
+
+    const reportPreviewActions = {
+        [CONST.REPORT.REPORT_PREVIEW_ACTIONS.SUBMIT]: (
+            <Button
+                success={isWaitingForSubmissionFromCurrentUser}
+                text={translate('common.submit')}
+                onPress={() => iouReport && submitReport(iouReport)}
```

Comment:
I think we can simplify this since we'd only land on the `Submit` action if we have an iouReport, right?
```suggestion
                onPress={() => submitReport(iouReport)}
```

Reply:
yes it's true but the function definition of submitReport requires so that its not null/undefined, without this check eslint screams